libcudacxx/include/cuda/std/__bit/clz.h

```diff
@@ -30,123 +30,77 @@
 
 _LIBCUDACXX_BEGIN_NAMESPACE_STD
 
-_LIBCUDACXX_HIDE_FROM_ABI constexpr int __binary_clz2(uint64_t __x, int __c)
-{
-  return !!(~__x & 0x2) ^ __c;
-}
-_LIBCUDACXX_HIDE_FROM_ABI constexpr int __binary_clz4(uint64_t __x, int __c)
-{
-  return __binary_clz2(__x >> 2 * !!(__x & 0xC), __c + 2 * !(__x & 0xC));
-}
-_LIBCUDACXX_HIDE_FROM_ABI constexpr int __binary_clz8(uint64_t __x, int __c)
-{
-  return __binary_clz4(__x >> 4 * !!(__x & 0xF0), __c + 4 * !(__x & 0xF0));
-}
-_LIBCUDACXX_HIDE_FROM_ABI constexpr int __binary_clz16(uint64_t __x, int __c)
-{
-  return __binary_clz8(__x >> 8 * !!(__x & 0xFF00), __c + 8 * !(__x & 0xFF00));
-}
-_LIBCUDACXX_HIDE_FROM_ABI constexpr int __binary_clz32(uint64_t __x, int __c)
-{
-  return __binary_clz16(__x >> 16 * !!(__x & 0xFFFF0000), __c + 16 * !(__x & 0xFFFF0000));
-}
-_LIBCUDACXX_HIDE_FROM_ABI constexpr int __binary_clz64(uint64_t __x)
-{
-  return __binary_clz32(__x >> 32 * !!(__x & 0xFFFFFFFF00000000), 32 * !(__x & 0xFFFFFFFF00000000));
-}
-
-#if !_CCCL_COMPILER(MSVC)
-
 _LIBCUDACXX_HIDE_FROM_ABI constexpr int __constexpr_clz(uint32_t __x) noexcept
 {
-#  if defined(__CUDA_ARCH__)
-  return __binary_clz32(static_cast<uint64_t>(__x), 0); // no device constexpr builtins
-#  else
-  return __builtin_clz(__x);
-#  endif
+#if !_CCCL_COMPILER(MSVC) // workaround for GCC <= 9
+  return ::__builtin_clz(__x);
+#else // !_CCCL_COMPILER(MSVC) ^^^ / _CCCL_COMPILER(MSVC) vvv
```

Comment:
This seems not consistent with the comment.

Generally speaking we have moved a ton of the builtin handling into `__cccl/builtin.h` because that gives us a single source of truth regarding compiler support.